app/models/ems_event.rb

```diff
@@ -136,6 +137,18 @@ def self.process_host_in_event!(event, options = {})
     end
   end
 
+  def self.process_physical_storage_in_event!(event, options = {})
+    physical_storage_ems_ref = event[:physical_storage_ems_ref]
+    process_object_in_event!(PhysicalStorage, event, options)
+
+    if event[:physical_storage_id].nil? && physical_storage_ems_ref.present?
+      storage = PhysicalStorage.find_by(:ems_ref => physical_storage_ems_ref)
+      unless storage.nil?
+        event[:physical_storage_id] = storage.id
+      end
+    end
```

Comment:
Ideally this method would look like :point_down: 
```suggestion
    process_object_in_event!(PhysicalStorage, event, options)
```

Reply:
I don't send options object to "process_object_in_event!" so I think this method doesn't do anything useful

Reply:
I'll check it out to be sure

Reply:
That should be okay because it should use the `PhysicalStorage` to use the right defaults: https://github.com/Autosde/manageiq/blob/8f8b88154663751d11bb058a95f5620dbc86ce5a/app/models/ems_event.rb#L95

```ruby
>> prefix = nil
>> key_prefix = PhysicalStorage.name.underscore
=> "physical_storage"
>> id_key = "#{prefix}#{key_prefix}_id".to_sym
=> :physical_storage_id
>> ems_ref_key = "#{prefix}#{key_prefix}_ems_ref".to_sym
=> :physical_storage_ems_ref
>> name_key = "#{prefix}#{key_prefix}_name".to_sym
=> :physical_storage_name
```

Reply:
oh ok, let me check it